### PR TITLE
Drupal: Added a second link to rules-and-policies 

### PR DESCRIPTION
**Pull request by drshawnkwang**
so that if either exists, the Join Page will link to it.

https://dev.gridrepublic.org/browse/DBOINCP-393

### CODE REVIEW

File: drupal/sites/default/boinc/modules/boincuser/boincuser.module [hunk 1]
```diff
@@ -1201,7 +1201,12 @@ function boincuser_edit_profile($account = null) {
   */
 function join_page($type = null) {
   global $base_url;
-  $ruleslink = '/content/rules-and-policies';
+  /* The paths/links to the rules-and-policies page is hardcoded
+   * here. An improvement would be admin settings for the Join Page
+   * where this path could be set.
+   */
+  $ruleslinkA = '/rules-and-poilicies';
```
Comment (ChristianBeer) [inline drupal/sites/default/boinc/modules/boincuser/boincuser.module]: typo

Comment (drshawnkwang) [inline drupal/sites/default/boinc/modules/boincuser/boincuser.module]: Thanks!

File: drupal/sites/default/boinc/modules/boincuser/boincuser.module [hunk 2]
```diff
@@ -1241,9 +1246,14 @@ function join_page($type = null) {
...
-    if (menu_valid_path(array('link_path' => $ruleslink))) {
+    if ( menu_valid_path(array('link_path' => $ruleslinkA)) ) {
       $output .= '  <li>' . bts("Read our !rules_and_policies.", array(
-        '!rules_and_policies' => l(bts('Rules and Policies', array(), NULL, 'boinc:join-page'), $ruleslink),
+        '!rules_and_policies' => l(bts('Rules and Policies', array(), NULL, 'boinc:join-page'), $ruleslinkA),
+      ), NULL, 'boinc:join-page') . '</li>';
+    }
+    else if (menu_valid_path(array('link_path' => $ruleslinkB))) {
+      $output .= '  <li>' . bts("Read our !rules_and_policies.", array(
+        '!rules_and_policies' => l(bts('Rules and Policies', array(), NULL, 'boinc:join-page'), $ruleslinkB),
```
Comment (ChristianBeer) [inline drupal/sites/default/boinc/modules/boincuser/boincuser.module]: I don't like that there are now two identical `bts()` calls. What about rewriting it like this:
```
$ruleslink="";
if ( menu_valid_path(array('link_path' => $ruleslinkA)) ) {
  $ruleslink = $ruleslinkA;
} else if (menu_valid_path(array('link_path' => $ruleslinkB)) {
  $ruleslink = $ruleslinkB;
}
if ($ruleslink) {
output .= ... bts(...) ...;
}

**Comment by brevilo** [comment]
Please hold off merging this one until the original issue's discussion is settled 😁

File: drupal/sites/default/boinc/modules/boincuser/boincuser.module (hunk 2 above)
Comment (brevilo) [inline drupal/sites/default/boinc/modules/boincuser/boincuser.module]: I agree with Christian.

Comment (drshawnkwang) [inline drupal/sites/default/boinc/modules/boincuser/boincuser.module]: Should be fixed with commit 8e56db0

File: drupal/sites/default/boinc/modules/boincuser/boincuser.module [hunk 3]
```diff
@@ -1246,14 +1246,16 @@ function join_page($type = null) {
...
     if ( menu_valid_path(array('link_path' => $ruleslinkA)) ) {
-      $output .= '  <li>' . bts("Read our !rules_and_policies.", array(
-        '!rules_and_policies' => l(bts('Rules and Policies', array(), NULL, 'boinc:join-page'), $ruleslinkA),
-      ), NULL, 'boinc:join-page') . '</li>';
+      $ruleslink = $ruleslinkA;
+    } else if ( menu_valid_path(array('link_path' => $ruleslinkB)) ) {
+      $ruleslink = $ruleslinkB;
     }
-    else if (menu_valid_path(array('link_path' => $ruleslinkB))) {
+    dd($ruleslink, 'ruleslink');
```
Comment (tristanolive) [inline drupal/sites/default/boinc/modules/boincuser/boincuser.module]: The dd() function has to be removed for production release.

File: drupal/sites/default/boinc/modules/boincuser/boincuser.module [hunk 4]
```diff
@@ -1205,8 +1205,8 @@ function join_page($type = null) {
    * here. An improvement would be admin settings for the Join Page
    * where this path could be set.
    */
-  $ruleslinkA = '/rules-and-policies';
-  $ruleslinkB = '/content/rules-and-policies';
+  $ruleslinkA = $base_url . '/rules-and-policies';
```
Comment (tristanolive) [inline drupal/sites/default/boinc/modules/boincuser/boincuser.module]: This actually breaks the menu_valid_path() functionality, which always returns TRUE for absolute URLs. If $base_url is needed, it should be added just prior to rendering the link (though the l() function should handle this ok, too).

**Pull request merged**